lib/board.rb

```diff
@@ -29,4 +29,50 @@ def valid_coordinate?(coordinate)
       true
     end
   end
+
+  def valid_placement?(ship, coordinates)
+      if ship.length == coordinates.length && coordinates.all? { |coordinate| valid_coordinate? (coordinate) }
+        if place_ship?(coordinates) == true && is_occupied?(coordinates) == false
+          true
+        else
+          false
+        end
+      else
```

Comment:
I don't think we need this second else statement.

Reply:
If it's taken out, test_returns_valid_placement_length returns nil instead of false.